events/wc_undead_events.txt

```diff
@@ -1632,103 +1617,337 @@ character_event = {
 
+	is_triggered_only = yes
+	
+	option = {
+		name = EVTOPTA_WCUND_411
+		
+		remove_slothful_or_add_diligent_silently_effect = yes
+		
+		73 = { province_event = { id = WCUND.524 days = 30 } }
+	}
+}
+### Merges with Lich King
+# Delayer event
+province_event = {
+	id = WCUND.524
+	
+	is_triggered_only = yes
+	hide_window = yes
+	
 	immediate = {
-		e_scourge = { holder_scope = { save_event_target_as = target_lich_king } }
-		set_global_flag = merging_with_the_lich_king
+		# If... these conditions are met, event chain can go on
+		if = {
+			limit = {
+				event_target:target_lich_king = { merging_lich_king_trigger = yes }
+				event_target:target_champion = { merging_champion_trigger = yes }
+			}
+			event_target:target_champion = { narrative_event = { id = WCUND.525 } }
+		}
+		# Else... notify that event chain failed
+		else = {
+			event_target:target_lich_king = { character_event = { id = WCUND.540 } }
+			event_target:target_champion = { character_event = { id = WCUND.540 } }
+		}
 	}
+}
+# Delayed event
+narrative_event = {
+	id = WCUND.525
+	title = EVTTITLE_WCUND_400
+	desc = EVTDESC_WCUND_525
+	picture = GFX_evt_icecrown
+	border = GFX_event_narrative_frame_intrigue
+
+	is_triggered_only = yes
 
+	# Consume Lich King's soul - if Lich King is player, Lich King gets game over :3
 	option = {
-		name = EVTOPTA_WCUND_500
-		narrative_event = { id = WCUND.550 days = 30 }
+		trigger = {
+			event_target:target_lich_king = { ai = no }
+		}
+		name = EVTOPTA_WCUND_525
+		
+		hidden_effect = {
+			if = {
+				limit = { OR = { is_republic = yes is_theocracy = yes is_nomadic = yes } }
+				abdicate = yes
+			}
+		}
+		event_target:target_lich_king = {
+			show_scope_change = no
+
+			hidden_effect = {
+				ROOT = { wealth = PREV }
+				clear_wealth = yes
+			}
+			abdicate_to = ROOT
+			hidden_effect = {
+				death = { death_reason = death_merged killer = ROOT }
+			}
+		}
+		
+		ai_chance = {
+			factor = 0
+		}
 	}
+	
+	# Merge with Lich King's soul
 	option = {
-		name = EVTOPTB_WCUND_500
+		name = EVTOPTB_WCUND_525
+		
+		hidden_effect = {
+			if = {
+				limit = { OR = { is_republic = yes is_theocracy = yes is_nomadic = yes } }
+				abdicate = yes
+			}
+		}
+		event_target:target_lich_king = {
+			show_scope_change = no
+			
+			hidden_effect = {
+				ROOT = {
+					create_temporary_landless_title_effect = yes	# Creates temporary landless title to avoid game over
+					wealth = PREV
+				}
+				set_player_character = ROOT	# Current Lich King starts playing as ROOT
+				clear_wealth = yes
+			}
+			abdicate_to = ROOT
+			hidden_effect = {
+				ROOT = { destroy_temporary_landless_title_effect = yes }	# Destroys title created in create_temporary_landless_title_effect
+				death = { death_reason = death_merged killer = ROOT }
+			}
+		}
+	}
+	
+	after = {
+		remove_trait = travelling
+		add_trait = sleeping_trait
+		
+		narrative_event = { id = WCUND.530 years = 2 }
+	}
+}
+### Visions
+narrative_event = {
+	id = WCUND.530
```

Comment:
Fires twice

<details>
<summary>Click to expand</summary>

![image](https://user-images.githubusercontent.com/46576860/71331324-b603e800-2542-11ea-8d87-591e911d4904.png)

</details>

Reply:
Soon... how to reproduce it?
<details>
<summary>Click to expand</summary>

![image](https://user-images.githubusercontent.com/46384992/71366231-e2f3e180-25ba-11ea-9f1a-94ff7a99c942.png)

</details>

Reply:
How you managed to not reproduce it?

<details>
<summary>Click to expand</summary>

![image](https://user-images.githubusercontent.com/46576860/71393885-cc33a600-261f-11ea-983c-fdaed7029ace.png)


</details>

Reply:
Played Arthas, merged with the Lich King, waited, one event fired.
<details>
<summary>Click to expand</summary>

![image](https://user-images.githubusercontent.com/46384992/71449780-dfed2280-276d-11ea-9016-d82b23ae50c0.png)
![image](https://user-images.githubusercontent.com/46384992/71449793-204ca080-276e-11ea-84b8-64d29ed2f108.png)

</details>

Reply:
Try to play as Ner'zhul

Reply:
Got it. Fixed in https://github.com/Warcraft-GoA-Development-Team/Warcraft-Guardians-of-Azeroth/commit/df22bd52664f753c447c0e41453ac682f9719a9a.